### DISCUSSION TITLE
I want to display a custom object on the timeline

### DISCUSSION THREAD

**Discussion by h-yorifuji** [deejay-hub/timeline-lwc#300] [Q&A] [answered]
I would like to display a custom object that has a reference relationship with a contact on the timeline, but this is not possible.
I've read the documentation and other Q&As, but I can't figure out the problem.
Please tell me the solution.

![オブジェクト](https://github.com/user-attachments/assets/5a6105fa-7996-4bdf-af93-e021ed3f2213)
![項目](https://github.com/user-attachments/assets/380279d5-6fd2-46b5-9c0a-c3b0f30b59d0)
![カスタムメタデータ](https://github.com/user-attachments/assets/989ce6da-d789-495d-bf1a-ea6c491ce81e)

**Comment by deejay-hub** [comment]
If I understand correctly looks like you created a custom object called KazokuJoho__c.

Then you added a lookup or master/detail relationship to Contact using a custom field called KazokuContact__c. When you created the field you picked the relationship name **KazokuJohos2** NOTE THAT YOU ADDED THE NUMBER 2 as highlighted

<img width="1557" alt="image" src="https://github.com/user-attachments/assets/c841d5f1-b833-4f98-8cd8-aad67e913a36">

This field tells Salesforce what the name of the relationship becomes.

So in your case the Relationship Name field in the metadata - in Timeline Configuration should be **KazokuJohos2__r**.

Let us know if that helps.

**Comment by h-yorifuji** [comment]
Thank you for your reply.
I uploaded the wrong image.
This is the correct one.

![項目情報](https://github.com/user-attachments/assets/a72e88d5-41e1-4bb9-83ef-41fb02b308eb)
So I think the Relationship Name field is correct.
Are there any other places I'm wrong?

**Reply by deejay-hub** [reply]
Well off the top of my head I would check the following:

- That you have permissions to the object and fields listed in the metadata
- That the field SeinenGappi__c has a value for records you expect to appear and that the value is within the time range specified when adding Time Warp to the page (e.g Time in Future, Time in past)
- that the field SeinenGappi__c is a date or date/time field (or a formula that returns that data type)
- that the record is not hidden. For larger data sets the record might be off the visible range requiring you to scroll down to see it.
- that the field Contact__c has a Contact record populated for the parent you are using

Tomorrow I'll recreate the setup you have and upload some images to help.

**Comment by h-yorifuji** [comment]
I confirmed what you pointed out.
When I check the time warp that I added to the contact page, in addition to not seeing any records, the filter also doesn't include the KazokuJohos__ｃ.
I'm waiting for your answer tomorrow.

**Comment by deejay-hub** [comment, marked as answer]
Step 1: I created a new custom object
<img width="1251" alt="image" src="https://github.com/user-attachments/assets/c6a6bd55-287d-4a89-bb9f-39051b3e92db">

Step 2: I created 3 new custom fields
SeinenGappi__c / Date
KazokuOkyakusamaMei__c (Text255)
Contact__c (Lookup to Contact) / Child Relationship Name - KazokuJohos
<img width="1259" alt="image" src="https://github.com/user-attachments/assets/bd325ac0-ba71-4d50-9b88-6e25f751c1ce">

Step 3: Setup custom metadata
<img width="989" alt="image" src="https://github.com/user-attachments/assets/d5667685-69b2-41de-adbf-0e26c5cd5be3">

Step 4: Logged out and back in again

Step 5: Created a new related record for Rachel Morris (Contact)
<img width="1259" alt="image" src="https://github.com/user-attachments/assets/31b45f94-712c-4297-a996-60c15ef92732">

Step 5: Added Time Warp the page
<img width="1262" alt="image" src="https://github.com/user-attachments/assets/305f7614-689d-4b75-8d71-d923f0659c06">

Step 6: Update metadata with custom fields for position date and display field
<img width="998" alt="image" src="https://github.com/user-attachments/assets/cda18168-8309-473b-b885-14114fff961f">

Step 7: Verify record still displays using new values
<img width="1263" alt="image" src="https://github.com/user-attachments/assets/9d8e0e5f-58e5-4d4b-bb47-33cd229216bb">

Since Salesforce caches records please try logging out and back in. If your setup is any different then let us know.

We are using Time Warp v1.14
<img width="793" alt="image" src="https://github.com/user-attachments/assets/f5b3c5f5-bd09-4a3b-bff0-a2e0d0a94081">

**Comment by h-yorifuji** [comment]
Thank you for your help.
By recreating the custom metadata, I was able to make the custom object appear in the milestone!
